### PR TITLE
Added benchmark test and convergence test for the NCF model

### PR DESCRIPTION
**Pull request by seemuch**

### CODE REVIEW

File: official/recommendation/ncf_keras_benchmark.py [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+    if root_data_dir:
+      FLAGS.data_dir = os.path.join(root_data_dir, 'movielens_data')
+
+  def _setup(self):
+    """Sets up and resets flags before each test."""
+    tf.compat.v1.logging.set_verbosity(tf.compat.v1.logging.DEBUG)
+    if KerasNCFBenchmarkBase.local_flags is None:
+      # Loads flags to get defaults to then override. List cannot be empty.
+      flags.FLAGS(['foo'])
+      # Overrides flag values with defaults for the class of tests.
```
Comment (robieta) [inline official/recommendation/ncf_keras_benchmark.py]: There are methods for manual flag setting in util.
https://github.com/tensorflow/models/blob/master/official/utils/flags/core.py#L38:#L46

Comment (seemuch) [reply official/recommendation/ncf_keras_benchmark.py]: Done.

File: official/recommendation/ncf_keras_benchmark.py [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+        **kwargs)
+
+  def _extract_benchmark_report_extras(self, stats):
+    extras = {}
+    extras['train_loss'] = stats['loss']
+    extras['eval_hit_rate'] = stats['eval_hit_rate']
+    extras['examples_per_second'] = stats['avg_exp_per_second']
+    return extras
+
+  def benchmark_ncf(self):
```
Comment (tfboyd) [inline official/recommendation/ncf_keras_benchmark.py]: r/benchmark_ncf/benchmark_1_gpu

File: official/recommendation/ncf_keras_benchmark.py [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+        output_dir=output_dir,
+        default_flags=default_flags,
+        **kwargs)
+
+  def _extract_benchmark_report_extras(self, stats):
+    extras = {}
+    extras['examples_per_second'] = stats['avg_exp_per_second']
+    return extras
+
+  def benchmark_ncf(self):
```
Comment (tfboyd) [inline official/recommendation/ncf_keras_benchmark.py]: r/benchmark_ncf/benchmark_1_gpu

**Pull request merged**